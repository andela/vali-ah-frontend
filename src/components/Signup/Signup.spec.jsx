// react libraries
import React from 'react';
import { shallow } from 'enzyme';

import { SignupComponent } from './index';

describe('Signup', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      createUserRequest: jest.fn(),
      signup: { success: '', error: '', isRequesting: false }
    };
    wrapper = shallow(<SignupComponent
      createUserRequest={jest.fn()}
      signup={props.signup}
    />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('getDerivedStateFromProps', () => {
    it('should show div containing success message', () => {
      wrapper.setProps({
        ...props,
        signup: {
          ...props.signup,
          success: 'Success Message'
        }
      });
      const showMessageDiv = wrapper.find({ id: 'shwMessage' });

      expect(showMessageDiv.exists()).toBe(true);
    });
  });
  describe('Signup loader', () => {
    it('should show loader when request is true', () => {
      wrapper.setProps({
        ...props,
        signup: {
          ...props.signup,
          isRequesting: true
        }
      });
      const loaderWrapper = wrapper.find('.loader-wrapper');

      expect(loaderWrapper.exists()).toBe(true);
    });
  });
  describe('handleSubmit', () => {
    const state = {
      errors: {}
    };
    describe('if there are no errors', () => {
      const event = {
        preventDefault() {},
        target: { firstName: 'vali', lastName: 'vali' }
      };
      it('should call "createUserRequest" method', () => {
        wrapper.instance().handleInput(event);
        wrapper.setState({
          ...state
        });
        wrapper.find('form').simulate('onSubmit', { preventDefault: jest.fn() });

        expect(true).toBe(true);
      });
    });

    describe('if there are errors', () => {
      it('should call "createUserRequest" method', () => {
        wrapper.setState({
          ...state,
          errors: {
            ...state.errors,
            password: 'error in password'
          }
        });
        wrapper.find('form').simulate('onSubmit', { preventDefault: jest.fn() });

        expect(props.createUserRequest).not.toHaveBeenCalled();
      });
    });

    describe('', () => {
      const event = {
        preventDefault() {},
        target: {
          firstName: 'vali',
          lastName: 'vali',
          userName: 'vali',
          email: 'vali@gmail.com',
          password: 'p',
          confirmPassword: 'password'
        }
      };
      it('should handle submit', () => {
        wrapper.instance().handleSubmit(event);
        wrapper.setState({
          ...state,
          errors: {
            ...state.errors,
            password: 'error in password'
          }
        });
        wrapper.find('form').simulate('onSubmit', { preventDefault: jest.fn() });

        expect(props.createUserRequest).not.toHaveBeenCalled();
      });
    });
  });
});
