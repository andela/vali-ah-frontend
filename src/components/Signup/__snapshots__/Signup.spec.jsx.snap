// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup should match snapshot 1`] = `
<div>
  <form
    className="form-wrapper"
    onSubmit={[Function]}
  >
    <div
      className="input-fields-row"
    >
      <label
        className="form__label"
        htmlFor="firstName"
      >
        <Input
          className="label__input-element"
          name="firstName"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          value=""
        />
        <span
          className="form__errors"
        />
      </label>
      <label
        className="form__label"
        htmlFor="lastName"
      >
        <Input
          className="label__input-element"
          name="lastName"
          onChange={[Function]}
          placeholder="Last Name"
          required={true}
          value=""
        />
        <span
          className="form__errors"
        />
      </label>
    </div>
    <div
      className="input-fields-row"
    >
      <label
        className="form__label"
        htmlFor="userName"
      >
        <Input
          className="label__input-element"
          name="userName"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
        <span
          className="form__errors"
        />
      </label>
      <label
        className="form__label"
        htmlFor="email"
      >
        <Input
          className="label__input-element"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          value=""
        />
        <span
          className="form__errors"
        />
      </label>
    </div>
    <div
      className="input-fields-row"
    >
      <label
        className="form__label"
        htmlFor="password"
      >
        <Input
          className="label__input-element"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <span
          className="form__errors"
        />
      </label>
      <label
        className="form__label"
        htmlFor="password"
      >
        <Input
          className="label__input-element"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
          value=""
        />
        <span
          className="form__errors"
          id="shwMessage"
        />
      </label>
    </div>
    <div
      className="btn-section"
    >
      <Button
        className="form__submit-btn"
        id=""
        text="Signup"
        type="submit"
      >
        Signup
      </Button>
    </div>
  </form>
</div>
`;
